scripts/download-projects.py

```diff
@@ -148,7 +166,7 @@ def main(argv):
                     failed = True
                     continue
 
-            if not _clone_repo(project.repo_url, repo_dir):
+            if not (_clone_repo(project.repo_url, repo_dir) and _checkout_ref(repo_dir, version)):
```

Comment:
Not sure if I have read this correctly. Are we running `_checkout_ref` only when `_clone_repo` fails?

Reply:
Oh no, the `not` applies to both conditions. With extra whitespace, it looks like this:

```python
if not (
    _clone_repo(project.repo_url, repo_dir)
    and _checkout_ref(repo_dir, version)
):
```

Reply:
Gotcha